tests/unit/amo/components/TestAddAddonToCollection.js

```diff
@@ -0,0 +1,388 @@
+import React from 'react';
+
+import AddAddonToCollection, {
+  extractId, AddAddonToCollectionBase, mapStateToProps,
+} from 'amo/components/AddAddonToCollection';
+import {
+  addAddonToCollection,
+  addonAddedToCollection,
+  createInternalCollection,
+  fetchUserCollections,
+  loadUserCollections,
+} from 'amo/reducers/collections';
+import { ErrorHandler } from 'core/errorHandler';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  createFakeCollectionDetail,
+  dispatchClientMetadata,
+  dispatchSignInActions,
+  fakeAddon,
+} from 'tests/unit/amo/helpers';
+import {
+  createFakeEvent,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+import ErrorList from 'ui/components/ErrorList';
+
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata().store;
+  });
+
+  const getProps = (customProps = {}) => {
+    return {
+      addon: createInternalAddon(fakeAddon),
+      i18n: fakeI18n(),
+      store,
+      _window: {},
+      ...customProps,
+    };
+  };
+
+  const render = (customProps = {}) => {
+    const props = getProps(customProps);
+    return shallowUntilTarget(
+      <AddAddonToCollection {...props} />, AddAddonToCollectionBase
+    );
+  };
+
+  const signInAndDispatchCollections = ({
+    userId = 1,
+    collections = [
+      createFakeCollectionDetail({ authorId: userId }),
+    ],
+  } = {}) => {
+    dispatchSignInActions({ store, userId });
+    store.dispatch(loadUserCollections({ userId, collections }));
+  };
+
+  const createSomeCollections = ({ authorId = 1 } = {}) => {
+    const firstCollection = createFakeCollectionDetail({
+      authorId, name: 'first',
+    });
+    const secondCollection = createFakeCollectionDetail({
+      authorId, name: 'second',
+    });
+
+    return { firstCollection, secondCollection };
+  };
+
+  it('lets you specify the css class', () => {
+    const root = render({ className: 'MyClass' });
+
+    expect(root).toHaveClassName('MyClass');
+    expect(root).toHaveClassName('AddAddonToCollection');
+  });
+
+  describe('fetching user collections', () => {
+    it('fetches user collections on first render', () => {
```

Comment:
Should (can?) we have a test that makes sure this is only called on client-side render? (https://github.com/mozilla/addons-frontend/pull/4148/files#diff-79d1d6288c9a4baa1c738148e96b9b83R63)

Reply:
I don't think we should because there's very little chance of regressing the feature of loading only on the client. The feature relies on `componentDidMount()` which is well understood to be client-side only and React is committed to supporting that method for the foreseeable future.

Reply:
Right, true. Sorry 'bout that.